Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM python:3.12.0a7-buster as base
+
+# Install poetry
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH /root/.local/bin:$PATH
+
+# Make workspace
+WORKDIR /app
+COPY poetry.toml .
+COPY pyproject.toml .
+
+# Install poetry dependencies
+RUN poetry install
+
+# Copy production files
+FROM base as production
+
+COPY todo_app/templates/layout.html todo_app/templates/
+COPY todo_app/templates/prod_index.html todo_app/templates/index.html
```

Comment:
I'm wary of this sort of switch out; it's very easy to lose, or end up testing version A but releasing version B!
If we want a difference in behaviour, then I prefer using flags (e.g. environment variables) to change behaviour and always use as many of the same files as we can